File: backend/src/SportsGurukul.Domain/Entities/ContactInformation.cs
using SportsGurukul.Domain.Common;

namespace SportsGurukul.Domain.Entities;

public class ContactInformation : BaseEntity
{
    public Guid UserProfileId { get; set; }
    public string? PrimaryPhoneCountryCode { get; set; }
    public string? PrimaryPhoneNumber { get; set; }
    public bool PrimaryPhoneVerified { get; set; }
    public string? SecondaryPhoneCountryCode { get; set; }
    public string? SecondaryPhoneNumber { get; set; }
    public bool SecondaryPhoneVerified { get; set; }
    public string? WebsiteUrl { get; set; }
    public string? FacebookUrl { get; set; }
    public string? TwitterUrl { get; set; }
    public string? InstagramUrl { get; set; }
    public string? LinkedInUrl { get; set; }
    public string? YouTubeUrl { get; set; }

    public UserProfile UserProfile { get; set; } = null!;

    public PhoneNumber? GetPrimaryPhoneNumber()
    {
        if (string.IsNullOrEmpty(PrimaryPhoneCountryCode) || string.IsNullOrEmpty(PrimaryPhoneNumber))
            return null;

        return new PhoneNumber(PrimaryPhoneCountryCode, PrimaryPhoneNumber, PrimaryPhoneVerified);
    }

    public PhoneNumber? GetSecondaryPhoneNumber()
    {
        if (string.IsNullOrEmpty(SecondaryPhoneCountryCode) || string.IsNullOrEmpty(SecondaryPhoneNumber))
            return null;

        return new PhoneNumber(SecondaryPhoneCountryCode, SecondaryPhoneNumber, SecondaryPhoneVerified);
    }
}

File: mobile/lib/features/user/domain/entities/contact_information.dart
import 'package:flutter/foundation.dart';

/// Contact details associated with a user profile.
@immutable
class ContactInformation {
  const ContactInformation({
    required this.id,
    this.primaryPhoneCountryCode,
    this.primaryPhoneNumber,
    this.primaryPhoneVerified = false,
    this.secondaryPhoneCountryCode,
    this.secondaryPhoneNumber,
    this.secondaryPhoneVerified = false,
    this.websiteUrl,
    this.facebookUrl,
    this.twitterUrl,
    this.instagramUrl,
    this.linkedInUrl,
    this.youTubeUrl,
  });

  final String id;
  final String? primaryPhoneCountryCode;
  final String? primaryPhoneNumber;
  final bool primaryPhoneVerified;
  final String? secondaryPhoneCountryCode;
  final String? secondaryPhoneNumber;
  final bool secondaryPhoneVerified;
  final String? websiteUrl;
  final String? facebookUrl;
  final String? twitterUrl;
  final String? instagramUrl;
  final String? linkedInUrl;
  final String? youTubeUrl;

  @override
  bool operator ==(Object other) =>
      identical(this, other) ||
      other is ContactInformation &&
          runtimeType == other.runtimeType &&
          id == other.id &&
          primaryPhoneCountryCode == other.primaryPhoneCountryCode &&
          primaryPhoneNumber == other.primaryPhoneNumber &&
          primaryPhoneVerified == other.primaryPhoneVerified;

  @override
  int get hashCode => Object.hash(
    id,
    primaryPhoneCountryCode,
    primaryPhoneNumber,
    primaryPhoneVerified,
  );
}
